turbo/jsonrpc/bor_snapshot.go

```diff
@@ -193,6 +199,93 @@ func (api *BorImpl) GetCurrentValidators() ([]*valset.Validator, error) {
 	return snap.ValidatorSet.Validators, nil
 }
 
+func (api *BorImpl) SendRawTransactionConditional(ctx context.Context, encodedTx hexutility.Bytes, options types.OptionsAA4337) (common.Hash, error) {
+	txn, err := types.DecodeWrappedTransaction(encodedTx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	// this has been moved to prior to adding of transactions to capture the
+	// pre state of the db - which is used for logging in the messages below
+	tx, err := api.db.BeginRo(ctx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	defer tx.Rollback()
+
+	currentHeader := rawdb.ReadCurrentHeader(tx)
+	// Ensure we have an actually valid block
+	if currentHeader == nil {
+		return common.Hash{}, errUnknownBlock
+	}
+
+	tempBlockNumber := rpc.BlockNumber(currentHeader.Number.Int64())
+	tempBlockorHash := rpc.BlockNumberOrHash{BlockNumber: &tempBlockNumber}
+
+	readerTemp, err := rpchelper.CreateStateReader(ctx, tx, tempBlockorHash, 0, api.filters, api.stateCache, api.historyV3(tx), "")
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	currentState := state.New(readerTemp)
+
+	// check block number range
+	if err := currentHeader.ValidateBlockNumberOptions4337(options.BlockNumberMin, options.BlockNumberMax); err != nil {
+		return common.Hash{}, &rpc.OptionsValidateError{Message: "out of block range. err: " + err.Error()}
+	}
+
+	// check timestamp range
+	if err := currentHeader.ValidateTimestampOptions4337(options.TimestampMin, options.TimestampMax); err != nil {
+		return common.Hash{}, &rpc.OptionsValidateError{Message: "out of time range. err: " + err.Error()}
+	}
+
+	// check knownAccounts length (number of slots/accounts) should be less than 1000
+	if err := options.KnownAccounts.ValidateLength(); err != nil {
```

Comment:
we could reuse the options validation function here (see my comment in stage_mining_exec.go)

Reply:
Done [here](https://github.com/ledgerwatch/erigon/pull/8229/commits/254e2884e95530f69319326449211e2ae30af6c1).